src/backend/libs/efiling-account-client/src/main/java/ca/bc/gov/open/jag/efilingaccountclient/CsoAccountServiceImpl.java

```diff
@@ -7,44 +7,50 @@
 import ca.bc.gov.ag.csows.accounts.ClientProfile;
 import ca.bc.gov.ag.csows.accounts.NestedEjbException_Exception;
 import ca.bc.gov.open.jag.efilingaccountclient.exception.CSOHasMultipleAccountException;
+import ca.bc.gov.open.jag.efilingaccountclient.mappers.AccountDetailsMapper;
+import ca.bc.gov.open.jag.efilingcommons.model.AccountDetails;
 import org.springframework.util.StringUtils;
 
 import java.util.List;
 
 public class CsoAccountServiceImpl implements EfilingAccountService {
 
-    private AccountFacadeBean accountFacadeBean;
-    private RoleRegistryPortType roleRegistryPortType;
+    private final AccountFacadeBean accountFacadeBean;
+    private final RoleRegistryPortType roleRegistryPortType;
+    private final AccountDetailsMapper accountDetailsMapper;
 
-    public CsoAccountServiceImpl(AccountFacadeBean accountFacadeBean, RoleRegistryPortType roleRegistryPortType) {
+    public CsoAccountServiceImpl(AccountFacadeBean accountFacadeBean, RoleRegistryPortType roleRegistryPortType,
+                                 AccountDetailsMapper accountDetailsMapper) {
 
         this.accountFacadeBean = accountFacadeBean;
         this.roleRegistryPortType = roleRegistryPortType;
+        this.accountDetailsMapper = accountDetailsMapper;
     }
 
     @Override
-    public CsoAccountDetails getAccountDetails(String userGuid) throws NestedEjbException_Exception {
+    public AccountDetails getAccountDetails(String userGuid) throws NestedEjbException_Exception {
 
         if (StringUtils.isEmpty(userGuid)) return null;
 
-        CsoAccountDetails csoAccountDetails = null;
+        AccountDetails accountDetails = null;
         List<ClientProfile> profiles = accountFacadeBean.findProfiles(userGuid);
         //An account must only one profile associated to proceed
         if (profiles.size() == 1) {
-            ClientProfile profile = profiles.get(0);
-            csoAccountDetails = new CsoAccountDetails(profile.getAccountId(), profile.getClientId(), HasFileRole(userGuid));
+            accountDetails = accountDetailsMapper.toCsoAccountDetails(profiles.get(0), hasFileRole(userGuid));
         }
         else if (profiles.size() > 1) {
             throw new CSOHasMultipleAccountException(profiles.get(0).getClientId().toString());
         }
 
-        return csoAccountDetails;
+        return accountDetails;
+
     }
 
-    public boolean HasFileRole(String userGuid) {
+    public boolean hasFileRole(String userGuid) {
```

Comment:
question? cam/should this be `isEfileRolePresent()`

Reply:
actually changed it to `isFileRolePresent` the role is `FILE` in cso